tests/Jellyfin.Model.Tests/Dlna/StreamInfoTests.cs

```diff
@@ -0,0 +1,236 @@
+    /// <returns>The string.</returns>
+    private static string RandomString(Random random, int minLength = 0, int maxLength = 256)
+    {
+        var len = random.Next(minLength, maxLength);
+        var sb = new StringBuilder(len);
+
+        while (len > 0)
+        {
+            sb.Append((char)random.Next(65, 97));
+            len--;
+        }
+
+        return sb.ToString();
+    }
+
+    /// <summary>
+    /// Creates a random long.
+    /// </summary>
+    /// <param name="random">The <see cref="Random"/> instance.</param>
+    /// <param name="min">Min value.</param>
+    /// <param name="max">Max value.</param>
+    /// <returns>A random <see cref="long"/> between <paramref name="min"/> and <paramref name="max"/>.</returns>
+    private static long RandomLong(Random random, long min = -9223372036854775808, long max = 9223372036854775807)
+    {
+        long result = random.Next((int)(min >> 32), (int)(max >> 32));
+        result <<= 32;
+        result |= (long)random.Next((int)(min >> 32) << 32, (int)(max >> 32) << 32);
+        return result;
+    }
+
+    /// <summary>
+    /// Creates a random string array containing between <paramref name="minLength"/> and <paramref name="maxLength"/>.
+    /// </summary>
+    /// <param name="random">The <see cref="Random"/> instance.</param>
+    /// <param name="minLength">The minimum number of elements.</param>
+    /// <param name="maxLength">The maximum number of elements.</param>
+    /// <returns>A random <see cref="string[]"/> instance.</returns>
+    private static string[] RandomStringArray(Random random, int minLength = 0, int maxLength = 9)
+    {
+        var len = random.Next(minLength, maxLength);
+        var arr = new List<string>(len);
+        while (len > 0)
+        {
+            arr.Add(RandomString(random, 1, 30));
+            len--;
+        }
+
+        return arr.ToArray();
+    }
+
+    /// <summary>
+    /// Creates a random int array containing between <paramref name="minLength"/> and <paramref name="maxLength"/>.
+    /// </summary>
+    /// <param name="random">The <see cref="Random"/> instance.</param>
+    /// <param name="minLength">The minimum number of elements.</param>
+    /// <param name="maxLength">The maximum number of elements.</param>
+    /// <returns>A random <see cref="int[]"/> instance.</returns>
+    private static int[] RandomIntArray(Random random, int minLength = 0, int maxLength = 9)
+    {
+        var len = random.Next(minLength, maxLength);
+        var arr = new List<int>(len);
+        while (len > 0)
+        {
+            arr.Add(random.Next());
+            len--;
+        }
+
+        return arr.ToArray();
+    }
+
+    /// <summary>
+    /// Fills most properties with random data.
+    /// </summary>
+    /// <param name="destination">The instance to fill with data.</param>
+    private static void FillAllProperties<T>(T destination)
+    {
+        var random = new Random();
```

Comment:
Randomness in a test isn't great.. should at least specify a seed so failures can be reproduced.

Reply:
I just copied them from the original PR. Still need to understand how they work...